manim/mobject/text/code_mobject.py

```diff
@@ -89,549 +64,186 @@ def construct(self):
+            code_string = code_file.read_text(encoding="utf-8")
+            lexer = guess_lexer_for_filename(code_file.name, code_string)
         else:
-            if self.insert_line_no:
-                foreground = VGroup(self.code, self.line_numbers)
+            if language is not None:
+                lexer = get_lexer_by_name(language)
             else:
-                foreground = self.code
-            height = foreground.height + 0.1 * 3 + 2 * self.margin
-            width = foreground.width + 0.1 * 3 + 2 * self.margin
+                lexer = guess_lexer(code_string)
 
-            rect = RoundedRectangle(
-                corner_radius=self.corner_radius,
-                height=height,
-                width=width,
-                stroke_width=self.background_stroke_width,
-                stroke_color=self.background_stroke_color,
-                color=self.background_color,
-                fill_opacity=1,
+        formatter = HtmlFormatter(
+            style=formatter_style,
+            noclasses=True,
+            cssclasses="",
+        )
+        soup = BeautifulSoup(
+            highlight(code_string, lexer, formatter), features="html.parser"
+        )
+        self._code_html = soup.find("pre")
+
+        # as we are using Paragraph to render the text, we need to find the character indices
+        # of the segments of changed color in the HTML code
+        color_ranges = []
+        current_line_color_ranges = []
+        current_line_char_index = 0
+        for child in self._code_html.children:
+            if child.name == "span":
+                try:
+                    color = child["style"].removeprefix("color: ")
+                except KeyError:
+                    color = None
+                current_line_color_ranges.append(
+                    (
+                        current_line_char_index,
+                        current_line_char_index + len(child.text),
+                        color,
+                    )
+                )
+                current_line_char_index += len(child.text)
+            else:
+                for char in child.text:
+                    if char == "\n":
+                        color_ranges.append(current_line_color_ranges)
+                        current_line_color_ranges = []
+                        current_line_char_index = 0
+                    else:
+                        current_line_char_index += 1
+
+        color_ranges.append(current_line_color_ranges)
+
+        code_lines = self._code_html.get_text().removesuffix("\n").split("\n")
+        self.code_lines = Paragraph(
+            *code_lines,
+            **paragraph_kwargs,
+        )
+        for line, color_range in zip(self.code_lines, color_ranges):
+            for start, end, color in color_range:
+                line[start:end].set_color(color)
+
+        if line_numbers:
+            paragraph_kwargs.update(
+                {
+                    "alignment": "right",
```

Comment:
Minor nitpick to get it to format better:
```suggestion
                    "alignment": "right"
```
so that it collapses it into one line

Reply:
True! e1fb43a5